includes/widgets/class-amp-widget-text.php

```diff
@@ -22,6 +22,7 @@ class AMP_Widget_Text extends WP_Widget_Text {
 		 * @return string $html The markup, unaltered.
 		 */
 		public function inject_video_max_width_style( $matches ) {
+			wp();
```

Comment:
This worries me a bit. Why do we need to run this manually in regular code?

Reply:
Because `is_amp_endpoint()` now warns of `_doing_it_wrong` if the `wp` action has not been done.

Reply:
Why is it worrisome?

Reply:
This function bootstraps WordPress, I think it shouldn't ever be called by anything but WordPress core itself. What is the problem here that might cause `wp()` to not have run before?

Reply:
It doesn't bootstrap WordPress. It just sets up the main WordPress query. 

Unit tests reset `$wp` after each test run:

https://github.com/WordPress/wordpress-develop/blob/67e12d960b166b12f7eee511bc8abf798533a872/tests/phpunit/includes/testcase.php#L162

And they essentially call `wp()` when calling the unit test `go_to()` method:

https://github.com/WordPress/wordpress-develop/blob/67e12d960b166b12f7eee511bc8abf798533a872/tests/phpunit/includes/testcase.php#L631-L637

Reply:
Right, but this instance here is not in a test, that's what gets me curious. Again, why is this necessary? What might cause `wp()` to not have run before a call to this?

Reply:
Oh, woops. You're right. I added it to the wrong place.

Reply:
Fixed: 744202d